Train model physics simulation

Jzang2011 opened the pull request:
## Description
Added support for the start of physical simulations. 
- Refactored the TrainModelUiController class to use javaFX properties and bindings.
- Added support to the TrainModelEnums to be used with ObjectProperties.
- Waiting for TrackModel Refactoring to add support for location. 

## Changes
- Added support for the start of physical simulations. 
- Refactored the TrainModelUiController class to use javaFX properties and bindings.
- Added support to the TrainModelEnums to be used with ObjectProperties.


## Testing
- Ran from TrainModelUserInterface Class. May need refactoring once further clarification of TrackModel is supplied

## Additional Information
N/A

Jzang2011 committed:
TrainModel physics simulation data

added fields for train specs.

Jzang2011 committed:
Merge branch 'master' into TrainModel-Physics-Simulation

Jzang2011 committed:
Simulation Dev

Added UnitConversions.java to utils module for later use.
Added base code for Simulations. Still needs further developement.

Jzang2011 committed:
Added Braking and Occupancy

Added brake() and update occupancy() to run function.

Jzang2011 committed:
Added Properties to Model class

Added properties to Model class to support binding with UI elements.
Added support in Enumerations for ObjectProperties for bindings.

Jzang2011 committed:
Fixed warning issue

fixed bug causing warning.

Jzang2011 committed:
Merge branch 'master' into TrainModel-Physics-Simulation

michaelplazek commented on TrainModel/src/trainmodel/TrainModelUserInterface.java:
```diff
@@ -1,55 +1,54 @@
   /**
    * Called by Main Menu to load the window for the TrainModel UI.
-   * TODO: Handle case of multiple instances of TrainModelUserInterface and TrainModelController
+   * TODO: Handle case of multiple instances of TrainModelUserInterface and TrainModelUiController
```
This should be TrainModelController

michaelplazek commented on TrainModel/src/trainmodel/view/TrainModelUiController.java:
```diff
@@ -185,42 +191,66 @@ private void end_failure_mode() {
+    trainModel = (TrainModel) TrainModelFactory
+        .createTrainModel(trainModelController,"001", "RED");
+  }
+
```
Won't this make every id == 001?

Jzang2011 replied:
It will. I still have to change that later to get the train ID from the trainController I believe.

Jzang2011 replied:
@michaelplazek I added commits to this PR to address those issues you had. I added TODO: comments in locations where I need to add development. I will include those in another PR if that works.

Jzang2011 replied:
Actually. I think this method will get refactored or completely deleted once I fully integrate my module with the CTC and MainMenu. Because Ideally those hard coded values will be given by the CTC. Right?

Jzang2011 committed:
TrainModelController renamed

Renamed TrainModelUIController to TrainModelController to meet agreed
naming convention.
Added comments for TODO’s

Jzang2011 committed:
Merge branch 'master' into TrainModel-Physics-Simulation

michaelplazek reviewed (changes requested):
Two small changes and then we can merge:
1. delete the "target" directory in TrainController module
2. Is the path to the TrainModelEnums correct for the TrainControllerModel for you? Have you tried running the program with your changes? I had to change the path there to run it. You SHOULD have to do the same. 

Once those are resolved I can merge.

michaelplazek reviewed (approved):
FIxed up!
